docs/developers/Backend.md

```diff
@@ -0,0 +1,125 @@
+initialization and finalization actors are optional.
+
+The following explanations of the traits mention multiple types defined in the Cromwell codebase. In particular
+`BackendJobDescriptor` wraps all the information necessary for a backend to instantiate a job and `JobKey` provides the
+information to uniquely identify a job. It is recommended that one look in the Cromwell codebase for more information on
+these and other types.
+
+#### BackendWorkflowInitializationActor
+
+If a backend developer wishes to take advantage of the initialization phase of the backend lifecycle they must implement
+this trait. There are three functions which must be implemented:
+
+1. `abortInitialization: Future[WorkflowAbortResponse]` specifies what to do, if anything, when a workflow is requested
+   to abort while a backend initialization is in progress.
+2. `validate: Future[Unit]` is provided so that the backend can ensure that all of the calls it will handle conform to
+   the rules of that backend. For instance, if a backend requires particular runtime attributes to exist.
+3. `beforeAll: Future[Unit]` is the actual initialization functionality. If a backend requires any work to be done prior
+   to handling a call, that code must be called from here.
+
+#### BackendJobExecutionActor
+
+There are three functions to override in this trait
+
+* `execute(jobDescriptor: BackendJobDescriptor): Future[BackendJobExecutionResponse]` requests that the backend run a
+  job. This will require detailed knowledge of how the execution platform works and the implementation of this function
+  will likely be unique to every platform. For instance an SGE backend using a shared filesystem might not need to do
+  anything with filename mappings and simply submit the job’s command to the SGE server. Another example would be the
+  [Google Life Sciences API](https://cloud.google.com/life-sciences/docs/reference/rest) which will spin up a VM with an
+  instance type determined from specified runtime attributes (# of CPU, amount of RAM), pull the task’s Docker image,
+  download the input files from Google Cloud Storage onto the VM, run the command in the Docker, copy the output files
+  up to Google Cloud Storage and shut down the VM.
+* `recover(jobDescriptor: BackendJobDescriptor): Future[BackendJobExecutionResponse]` requests that the backend
+  reconnect to a previously launched job. It is recommended that a backend developer implement this as requested but
+  that behavior isn’t appropriate for all situations, in which case it could for instance just call execute. In the case
+  of an SGE backend this might be implemented to determine the SGE ID for the job and start from there.
+* `abortJob(jobKey: BackendJobDescriptorKey): Future[JobAbortResponse]` requests that a running job be aborted. It is
+  recommended that this be properly implemented but that’s not always appropriate. This function could easily be
+  implemented as a no-op.
```

Comment:
These all look a little off compared to the code. Even the set of functions is a little different. For example, the description for `recover` here seems like it matches the actual `reconnect` function.